src/mantine-core/src/components/Overlay/Overlay.tsx

```diff
@@ -39,35 +42,58 @@ const defaultProps: Partial<OverlayProps<any>> = {
   color: '#fff',
   zIndex: getDefaultZIndex('modal'),
   radius: 0,
+  blur: 0,
 };
 
 export const Overlay: OverlayComponent = forwardRef(
   <C extends React.ElementType = 'div'>(props: OverlayProps<C>, ref: PolymorphicRef<C>) => {
-    const { opacity, color, gradient, zIndex, component, radius, sx, ...others } =
+    const { opacity, blur, color, gradient, zIndex, component, radius, sx, ...others } =
       useMantineDefaultProps('Overlay', defaultProps, props);
     const background = gradient ? { backgroundImage: gradient } : { backgroundColor: color };
 
-    return (
+    const baseStyles = {
+      position: 'absolute' as React.CSSProperties['position'],
+      top: 0,
+      bottom: 0,
+      left: 0,
+      right: 0,
+      zIndex,
+    };
+
+    const OverlayBox = (
```

Comment:
Do not start a variable name with a capital letter if not required. Rename variable to `innerOverlay`